client/src/holdings/AddHoldingDialog.tsx

```diff
@@ -0,0 +1,177 @@
+import './Dialog.css';
+
+import React, {useState} from 'react';
+import {Controller, useForm} from 'react-hook-form';
+import * as yup from 'yup';
+
+import {yupResolver} from '@hookform/resolvers/yup';
+import {
+    Button, CircularProgress, Dialog, DialogActions, DialogContent, DialogTitle, MenuItem, Select,
+    TextField
+} from '@material-ui/core';
+
+import {API} from '../api';
+import {CURRENCIES} from '../constants';
+import {CreateInstrumentData, Currency} from '../types';
+
+interface IFormInput {
+  name: string;
+  symbol: string;
+  category: string;
+  isin: string;
+  currency: Currency;
+  exchange: string;
+}
+
+type AddHoldingDialogProps = {
+  username: string;
+  onHoldingCreated: Function;
+  showNotification: Function;
+};
+
+/** Form to add a holding. */
+export function AddHoldingDialog({
+  username,
+  onHoldingCreated,
+  showNotification,
+}: AddHoldingDialogProps) {
+  const [open, setOpen] = useState(false);
+  const holdingValidationSchema = yup.object().shape({
+    name: yup.string(),
+    symbol: yup.string(),
+    category: yup.string(),
+    isin: yup.string(),
+    currency: yup.string(),
+    exchange: yup.string(),
+  });
+  const {
+    register,
+    handleSubmit,
+    control,
+    formState: {isSubmitting},
+  } = useForm<CreateInstrumentData>({
```

Comment:
According to the docs it should be the interface for the form in useForm<>: https://react-hook-form.com/api/#useForm

Rather than the one to send to the API. So can you replace this with IFormInput. It's weird how it's not complaining atm having the wrong type, maybe TS isn't working properly in this case, but good to have it as it's meant to be anyway.

Reply:
Fixed this and the other two